src/main/scala/com/rawlabs/das/server/DASSdkManager.scala

```diff
@@ -69,14 +72,28 @@ class DASSdkManager(implicit settings: RawSettings) extends StrictLogging {
             throw new IllegalArgumentException(s"DAS with id $dasId already registered")
           }
         case None =>
-          logger.debug(s"Registering DAS with ID: $dasId, Type: $dasType")
-          val dasSdk = dasSdkLoader
-            .find(_.dasType == dasType)
-            .getOrElse {
-              logger.error(s"DAS type '$dasType' not supported.")
-              throw new IllegalArgumentException(s"DAS type '$dasType' not supported")
-            }
-            .build(options)
+          // The client didn't specify a DAS ID. We generated a new one. We check if
+          // a DAS with the same options is already registered. Because if the client restarted, or is simply
+          // trying to register the same DAS again, we should reuse the existing DAS.
+
+          // Running DASes are indexed by their options. We only compare user visible options (those not starting with "das_")
+          val strippedOptions = options.filterKeys(!_.startsWith("das_"))
```

Comment:
You are stripping options in several places. Make a function?

Reply:
It's done in one place in the current changeset but we put it in a function to have an opportunity to comment the reason, and remove it if eventually we can.